src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1512,7 +1512,7 @@ def aks_addon_update(cmd, client, resource_group_name, name, addon, workspace_re
         if (instance.ingress_profile is None) or (instance.ingress_profile.web_app_routing is None) or not instance.ingress_profile.web_app_routing.enabled:
             raise InvalidArgumentValueError(f'Addon "{addon}" is not enabled in this cluster.')
 
-    if addon == "monitoring" and enable_msi_auth_for_monitoring is None:
+    elif addon == "monitoring" and enable_msi_auth_for_monitoring is None:
```

Comment:
@tg123, what's the problem you are trying to resolve? If you want to enable the Web Application Routing add-on on an existing cluster, please follow [doc](https://learn.microsoft.com/en-us/azure/aks/web-app-routing?tabs=without-osm) and use command `az aks enable-addons` instead.

Reply:
fix regression introduced by #6311 

original code 
```
if addon == "web_application_routing":
else:
   xxx
```

after 6311

```
if addon == "web_application_routing":

if addon == "monitoring":
else:
   xxx
```

which means when `addon == web_application_routing`, xxx will be always executed and raise ex

Reply:
OK, got it